kin-sdk-core/src/main/java/kin/sdk/core/KinAccountImpl.java

```diff
@@ -38,15 +41,23 @@
         this.keyStore = ethClientWrapper.getKeyStore();
         this.account = account;
         this.ethClient = ethClientWrapper;
+        hasDeleted = false;
     }
 
     @Override
     public String getPublicAddress() {
-        return account.getAddress().getHex();
+        if (!hasDeleted) {
+            return account.getAddress().getHex();
+        }
+        return "";
     }
 
     @Override
-    public String exportKeyStore(String passphrase, String newPassphrase) throws PassphraseException {
+    public String exportKeyStore(String passphrase, String newPassphrase)
+        throws PassphraseException, OperationFailedException {
+        if (hasDeleted) {
```

Comment:
preferably to wrap it in some `checkNotDeleted` method

Reply:
@yosriz  why enter to method stack instead of checking class member?